### DISCUSSION TITLE
How can I add a disallowed domain name for a client?

### DISCUSSION THREAD

**Discussion by metezd** [AdguardTeam/AdGuardHome#3879] [Q&A]
I want to disallow `google.com` domain name for client. When I enter `||www.google.com^$client='192.168.1.66'` it doesn't take any action. If I disallow `google.com` all the devices are disallowed so is there a syntax for disallowing it for a specific client?

**Comment by ainar-g** [comment]
Are you sure that your client has that IP and that it's not the IP of e.g. your router?  Do requests from this client—and only this client—show in the query log with this IP?

**Reply by metezd** [reply]
Yes, I can confirm that my client has that IP address. 
![client](https://user-images.githubusercontent.com/37701679/143767447-53f663b0-b08c-4dee-9fe8-605f63b7da9e.png)
Request from the client show in query log.
![image](https://user-images.githubusercontent.com/37701679/143767693-1ebe88dd-b1c0-4a47-aa90-8181a21e91bf.png)
Custom Filtering Rules does work but the requests overflow the query log so I want to disallow it
![disallowed](https://user-images.githubusercontent.com/37701679/143767638-2ecc0c28-d17e-4150-b210-317bcc099729.png)

**Reply by ainar-g** [reply]
Have you tried without the `www.` part?

**Reply by metezd** [reply]
Tried without `www.` didn't work. I am still trying some different ways to make it work

**Reply by ainar-g** [reply]
Wait, I just noticed.  You're not placing the rules into the right input.  These rules must be put into the _Custom filtering rules_ input.  In the _Filters_ category.

**Reply by metezd** [reply]
It was placed in _Custom Filtering Rules_ at first. Then I decided to move it to _Disallowed Domains_ because the query log was filled up with blocked `google.com` domain. I couldn't figure out how to manage it in _Disallowed Domains_

**Comment by EugeneOne1** [comment]
@metezd, we've just fixed #3914 within the latest edge build.  It may be related to the question so could you please install it and check if the requests are blocked?

**Reply by metezd** [reply]
Okay, I will try it and give a feedback soon

**Comment by metezd** [comment]
I've installed the latest beta version **v0.107.0-b.15**. _Custom Filtering Rules_ does work intended but _Disallowed Domains_ doesn't work. At this point I am going to stick with _Custom Filtering Rules_

**Reply by cookiemonsteruk** [reply]
I came looking for this answer, thank you metezd for asking and the adguard devs for a quick fix. I shall try and also post a related question separately.

**Comment by trenchcoatjedi** [comment]
I have been testing this as well on the latest edge build. I have entered ||www.linksys.com^$client='10.10.10.2' into both _Disallowed Domains_ and _Custom Filtering Rules_ since I have a linksys device which constantly tries to phone home and is polluting the logs.

This syntax in _Disallowed Domains_ does not prevent www.linksys.com from showing up as blocked for that specific client in the logs.

**Reply by EugeneOne1** [reply]
@trenchcoatjedi, the annotation to the *Disallowed Domains* field says:

 >  Not to be confused with filters. AdGuard Home drops DNS queries matching these domains, and these queries don't even appear in the query log. You can specify exact domain names, wildcards, or URL filter rules, e.g. "example.org", "*.example.org", or "||example.org^" correspondingly.

To clarify once more, it only works with domains (and URL rules to match these domains) without any additional context like clients or other rule modifiers.  You'd better use *Custom Filtering Rules* for client-specific blocking.
